src/ui.toit

```diff
@@ -290,37 +302,133 @@ class Ui:
 
+  /** Variant of $(emit-list --kind list) using the $DEBUG kind. */
+  emit-list --debug/bool list/List --title/string?=null:
+    emit-list --kind=DEBUG list --title=title
+
+  /** Variant of $(emit-list --kind list) using the $VERBOSE kind. */
+  emit-list --verbose/bool list/List --title/string?=null:
+    emit-list --kind=VERBOSE list --title=title
+
+  /** Variant of $(emit-list --kind list) using the $INFO kind. */
+  emit-list --info/bool list/List --title/string?=null:
+    emit-list --kind=INFO list --title=title
+
+  /** Variant of $(emit-list --kind list) using the $WARNING kind. */
+  emit-list --warning/bool list/List --title/string?=null:
+    emit-list --kind=WARNING list --title=title
+
+  /** Variant of $(emit-list --kind list) using the $INTERACTIVE kind. */
+  emit-list --interactive/bool list/List --title/string?=null:
+    emit-list --kind=INTERACTIVE list --title=title
+
+  /** Variant of $(emit-list --kind list) using the $ERROR kind. */
+  emit-list --error/bool list/List --title/string?=null:
+    emit-list --kind=ERROR list --title=title
+
+  /**
+  Variant of $(emit-list --kind list) using the $RESULT kind.
+
+  A program should do only a single result output per run.
+  */
+  emit-list --result/bool list/List --title/string?=null:
+    emit-list --kind=RESULT list --title=title
+
   /**
   Emits a map.
 
   Printers are *not* required to display the title.
   */
-  emit-map --kind/int=RESULT map/Map --title/string?=null:
+  emit-map --kind/int map/Map --title/string?=null:
     do_ --kind=kind:
       if printer_.needs-structured --kind=kind:
         printer_.emit-structured --kind=kind map
       else:
         printer_.emit-map --kind=kind --title=title map
 
+  /** Variant of $(emit-map --kind map) using the $DEBUG kind. */
+  emit-map --debug/bool map/Map --title/string?=null:
+    emit-map --kind=DEBUG map --title=title
+
+  /** Variant of $(emit-map --kind map) using the $VERBOSE kind. */
+  emit-map --verbose/bool map/Map --title/string?=null:
+    emit-map --kind=VERBOSE map --title=title
+
+  /** Variant of $(emit-map --kind map) using the $INFO kind. */
+  emit-map --info/bool map/Map --title/string?=null:
+    emit-map --kind=INFO map --title=title
+
+  /** Variant of $(emit-map --kind map) using the $WARNING kind. */
+  emit-map --warning/bool map/Map --title/string?=null:
+    emit-map --kind=WARNING map --title=title
+
+  /** Variant of $(emit-map --kind map) using the $INTERACTIVE kind. */
+  emit-map --interactive/bool map/Map --title/string?=null:
+    emit-map --kind=INTERACTIVE map --title=title
+
+  /** Variant of $(emit-map --kind map) using the $ERROR kind. */
+  emit-map --error/bool map/Map --title/string?=null:
+    emit-map --kind=ERROR map --title=title
+
+  /**
+  Variant of $(emit-map --kind map) using the $RESULT kind.
+
+  A program should do only a single result output per run.
+  */
+  emit-map --result/bool map/Map --title/string?=null:
```

Comment:
Usually we check that `result` is true in these cases.

Reply:
done.